src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -391,6 +391,13 @@ export class VisualizeEmbeddable
       syncColors: this.input.syncColors,
       uiState: this.vis.uiState,
       inspectorAdapters: this.inspectorAdapters,
+      executionContext: this.deps.start().core.executionContext.create({
+        type: this.vis.params.type,
+        name: this.vis.type.name,
+        id: this.vis.id ?? 'unknown_id',
```

Comment:
In what cases it may be empty? What should we use instead?

Reply:
unsaved visualization

Reply:
embeddable shouldn't be creating the execution context (as else we don't know if request came from dashboard or visualize for example) it should be the app that is responsible for this.

we should add execution context to EmbeddableInput (just as we did with searchSessionId)

Reply:
Should I fall back to `unsave_vis` instead of `unknown_id` then?

Reply:
as discussed in person, we will add this later. see `Nested context` in https://github.com/elastic/kibana/issues/102629